src/ocl/convolutionocl.cpp

```diff
@@ -1560,124 +956,37 @@ void ConvolutionDescriptor::FindConvBwdWeightsAlgorithm(Handle& handle,
-                                         : std::vector<miopen::solver::ConvSolution>{};
-            const auto winograd    = !miopen::IsDisabled(MIOPEN_DEBUG_CONV_WINOGRAD{})
-                                         ? FindWinogradWrWAllSolutions(ctx, problem, invoke_ctx)
-                                         : std::vector<miopen::solver::ConvSolution>{};
-            const auto implictgemm = !miopen::IsDisabled(MIOPEN_DEBUG_CONV_IMPLICIT_GEMM{})
-                                         ? FindImplicitGemmWrWAllSolutions(ctx, problem, invoke_ctx)
-                                         : std::vector<miopen::solver::ConvSolution>{};
-
-            // Precompile Solutions
-            {
-                std::vector<const miopen::solver::ConvSolution*> all;
-                all.reserve(gemm.size() + direct.size() + winograd.size() + implictgemm.size());
-                AppendPointersToElements(gemm, all);
-                AppendPointersToElements(direct, all);
-                AppendPointersToElements(winograd, all);
-                AppendPointersToElements(implictgemm, all);
-                PrecompileSolutions(handle, all);
-            }
-
-            // Evaluate Invokers
-            EvaluateInvokers(handle,
-                             gemm,
-                             AlgorithmName{"miopenConvolutionBwdWeightsAlgoGEMM"},
-                             network_config,
-                             invoke_ctx,
-                             record);
-            EvaluateInvokers(handle,
-                             direct,
-                             AlgorithmName{"miopenConvolutionBwdWeightsAlgoDirect"},
-                             network_config,
-                             invoke_ctx,
-                             record);
-            EvaluateInvokers(handle,
-                             winograd,
-                             AlgorithmName{"miopenConvolutionBwdWeightsAlgoWinograd"},
-                             network_config,
-                             invoke_ctx,
-                             record);
-            EvaluateInvokers(handle,
-                             implictgemm,
-                             AlgorithmName{"miopenConvolutionBwdWeightsAlgoImplicitGEMM"},
-                             network_config,
-                             invoke_ctx,
-                             record);
-        });
-    }
-
-    if(IsEnabled(MIOPEN_DEBUG_COMPILE_ONLY{}))
-        MIOPEN_THROW(miopenStatusGpuOperationsSkipped,
-                     "MIOPEN_DEBUG_COMPILE_ONLY is enabled, "
-                     "escaping backwards convolution. Search "
-                     "skipped.");
-
-    if(perf_db.empty())
+    const auto problem =
+        conv::ProblemDescription{dyDesc, dwDesc, xDesc, *this, conv::Direction::BackwardWeights};
+    const auto ctx = [&] {
+        auto tmp = ExecutionContext{&handle};
+        tmp.DetectRocm();
```

Comment:
I see this in many places, why cant this call be moved to the ctor ?

Reply:
I am not sure, but I'd rather do it in a separate PR.
But this is slow on the first call, it actually invokes stuff and caches the result in a static variable: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/ec1fb7ad54d22fe5d2934e3396c8bcc20375a049/src/ocl/gcn_asm_utils.cpp#L141-L141

Reply:
Perhaps we should move such initializations to the handle creation. The context is used every where in the library and it seems clunky to call this function everytime.